### PR TITLE
fix(nuxt): merge meta properties 

### LINKED ISSUE #28106: DefinePageMeta overwrites meta properties on NuxtPage objects.
### Environment

Nuxt project info:                                                                                            18:24:11

------------------------------
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.12.3
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      compatibilityDate, devtools, hooks
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-znsjbh?file=pages%2Ftest.vue

### Describe the bug

Context: The router is being generated by hooks:pages:extends, routes will have the meta property, and those pages will have the definePageMeta methods in them.
Issue: If definePageMeta is present on a vue page the meta property declared in the pages:extend hook will be overwritten instead of merged.
I can see this issue happening since 3.12.0

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by xjccc**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/28106

### CODE REVIEW

File: packages/nuxt/src/pages/utils.ts [hunk 1]
```diff
@@ -144,7 +144,11 @@ export async function augmentPages (routes: NuxtPage[], vfs: Record<string, stri
   for (const route of routes) {
     if (route.file && !augmentedPages.has(route.file)) {
       const fileContent = route.file in vfs ? vfs[route.file] : fs.readFileSync(await resolvePath(route.file), 'utf-8')
-      Object.assign(route, await getRouteMeta(fileContent, route.file))
+      let meta
+      if (route.meta || (await getRouteMeta(fileContent, route.file)).meta) {
+        meta = Object.assign({}, route.meta, (await getRouteMeta(fileContent, route.file)).meta)
+      }
```
Comment (huang-julien) [inline packages/nuxt/src/pages/utils.ts:150]:
```suggestion
      let meta = (await getRouteMeta(fileContent, route.file)).meta
      if (route.meta) {
        meta = Object.assign(meta, route.meta)
      }
```

**Review by danielroe — COMMENTED** [review]
would you be able to add a test?

I think we can also refactor this somewhat (perhaps with the spread operator?) so we aren't calling `getRouteMeta` twice.

**Comment by xjccc** [comment]
> would you be able to add a test?
> 
> 
> 
> I think we can also refactor this somewhat (perhaps with the spread operator?) so we aren't calling `getRouteMeta` twice.

Thanks. Let me have a try tomorrow. But i don't know what's the best way for this. Because I use tests less.😥

**Comment by xjccc** [comment]
@danielroe @huang-julien  Hi, brothers. I'v been add a test. But I don't know if it makes sense. Please help me for this. Thank you very much!😁

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Very nicely done - thank you 🙏❤️

**Pull request merged**